server/repository/src/migrations/views/mod.rs

```diff
@@ -153,7 +154,12 @@ pub(crate) fn rebuild_views(connection: &StorageConnection) -> anyhow::Result<()
         invoice.status AS invoice_status,
         invoice_line_stock_movement.total_before_tax AS total_before_tax,
         invoice_line_stock_movement.pack_size as pack_size,
-        invoice_line_stock_movement.number_of_packs as number_of_packs
+        invoice_line_stock_movement.number_of_packs as number_of_packs,
+        CASE
+          WHEN invoice.type IN ('INBOUND_SHIPMENT', 'CUSTOMER_RETURN', 'INVENTORY_ADDITION') THEN 1
+          WHEN invoice.type IN ('OUTBOUND_SHIPMENT', 'SUPPLIER_RETURN', 'PRESCRIPTION', 'INVENTORY_REDUCTION') THEN 2
+          ELSE 3
+        END AS type_precedence
```

Comment:
`type_precedence` was added to get stock_ins to calculate before stock_outs in case of somehow inbound and outbound ends up with the same datetime....